addons/stock/models/stock_move_line.py

```diff
@@ -241,36 +241,27 @@ def write(self, vals):
         moves_to_recompute_state = self.env['stock.move']
         Quant = self.env['stock.quant']
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
-        # We forbid to change the reserved quantity in the interace, but it is needed in the
-        # case of stock.move's split.
-        # TODO Move me in the update
-        if 'product_uom_qty' in vals:
-            for ml in self.filtered(lambda m: m.state in ('partially_available', 'assigned') and m.product_id.type == 'product'):
-                if not ml._should_bypass_reservation(ml.location_id):
-                    qty_to_decrease = ml.product_qty - ml.product_uom_id._compute_quantity(vals['product_uom_qty'], ml.product_id.uom_id, rounding_method='HALF-UP')
```

Comment:
Is it possible that `qty_to_decrease` was either negative or positive?

Reply:
Yes, it's possible (example: the reserved quantity was 20, I change it to 30, the qty_to_decrease is -10). But it was actually working since `_update_reserved_quantity` will reserve if `quantity` is positive and and unreserve if `quantity` is negative. So I think the original variable was wrongly named.

But we could check that vals['product_uom_qty'] is not negative, nothing good at all will happen if the user encodes that.